src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -2131,6 +2131,46 @@ private void checkInternetPermission() {
         }
     }
 
+    String serialize(final String uniqueUserId) throws AuthenticationException {
+        if (StringExtensions.IsNullOrBlank(uniqueUserId)) {
+            throw new IllegalArgumentException("uniqueUserId");
+        }
+        
+        if ((new BrokerProxy(mContext)).canSwitchToBroker()) {
+            throw new UsageAuthenticationException(ADALError.FAIL_TO_EXPORT,"Failed to export the FID because broker is enabled.");
+        }
+        
+        /* the current serialize/deserialize feature is for only supports MS apps 
+         * so the client ID for the FoCI token cache item is hard coded below
+         */        
+        final String cacheKey = CacheKey.createCacheKey(this.getAuthority(), null, AuthenticationConstants.MS_FAMILY_ID, true, uniqueUserId);
+        final TokenCacheItem tokenItem = this.getCache().getItem(cacheKey);
+        
+        if (tokenItem == null) {
+            Logger.i(TAG, "Cannot find the FoCI token cache item for this userID", "");
+            return null;
+        }        
+       
+        final SSOStateContainer blobContainer = new SSOStateContainer(tokenItem);
+        Logger.i(TAG, "prepare to serialize","");
+        return blobContainer.serialize();
+    }
+    
+    void deserialize(String serializedBlob) throws AuthenticationException {
+        if (StringExtensions.IsNullOrBlank(serializedBlob)) {
+            throw new IllegalArgumentException("serializedBlob");
+        }
+            
+        if ((new BrokerProxy(mContext)).canSwitchToBroker()) {
+            throw new UsageAuthenticationException(ADALError.FAIL_TO_IMPORT,"Failed to import the serialized blob because broker is enabled.");
+        }
+        
+        Logger.i(TAG, "prepare to deserialize","");
+        final TokenCacheItem tokenCacheItem = SSOStateContainer.deserialize(serializedBlob);
```

Comment:
1)  I don't like it, but if that is the idiom then we should certainly follow it.
2)  See comment below.  I'd rather rename this class to SSOStateSerializer and hide the fact that we have an extra container from any calling code.